### ISSUE TITLE
Data Handling

### ISSUE THREAD

**Issue by bpadovese** [bpadovese/DCASE-Task5-2022#1]
Hi @pmdnhd 

I had a look at the script you pushed, and it was very useful. It also highlighted that the dataset might be more challenging than what we expected.

I did some investigating and found that there is only 1 annotation with 0 duration. And it should be simple to remove those. simply adding this to end of the script (I will update it later):

```python
df_all.reset_index(drop=True, inplace=True)
df_all.drop(df_all[df_all.call_duration == 0].index, inplace=True)
```

I also checked how many overlapping annotations there are (only POS annotations) and there are not that many. And the ones that do overlap, it is only just a small portion at the end of one annotation and the start of another. So I dont think we need to be concerned about his.

Regardless. I have been thinking of ways to approach this dataset. The biggest issue as you pointed out is that there is a lot of variability both in terms of call duration and Sample rate used. I think it would be very challenging to use the raw audio or choosing just one set of parameters for a spectrogram that would work well for all species/calls.

So here is my initial suggestion. I think we could spend a little bit of time trying to study these species to group them into a couple of groups. This could be for instance High Frequency vocalizations, medium frequency and low frequency. With this, we could generate 3 spectrograms that would better to represent each of these groups (these spectrograms cant be vastly different from one another). We could also apply any sort of pre-processing steps like bandpass filters etc before spectrogram computation. Next, we train 3 models, each on one of these spectrograms. Final step would be to unify these models at some point. We could try an emsemble approach to this.

So in summary:

1. Brief study of the species involved
2. group species into 3 groups
3. Propose 3 spectrograms, where each spectrogram best represents one group
4. generate the 3 spectrograms for every sample
5. each of these spectrograms is fed into a model
6. The 3 models are combined through some emsemble technique.

What do you think? Have you already thought of something?

**Comment by pmdnhd** [comment]
Hi @bpadovese 

Great! Thanks for digging deeper on these stats, really helpful insights!

> Regardless. I have been thinking of ways to approach this dataset. The biggest issue as you pointed out is that there is a lot of variability both in terms of call duration and Sample rate used. I think it would be very challenging to use the raw audio or choosing just one set of parameters for a spectrogram that would work well for all species/calls.

Sounds good to me and let's keep in mind that we can also do 2-stage detect/classif to refine the DC

> So here is my initial suggestion. I think we could spend a little bit of time trying to study these species to group them into a couple of groups. This could be for instance High Frequency vocalizations, medium frequency and low frequency. With this, we could generate 3 spectrograms that would better to represent each of these groups (these spectrograms cant be vastly different from one another). We could also apply any sort of pre-processing steps like bandpass filters etc before spectrogram computation. Next, we train 3 models, each on one of these spectrograms. Final step would be to unify these models at some point. We could try an emsemble approach to this.

Sounds good too for a first approach. We will need to make the group assignment automated as a future step. Might be cool to investigate several representations (spectros, wavelets, ... ) to see which one fits best each group. I like the ensemble idea and this seems to be a good first method candidate! Let's start  with that!
